database-commons/src/main/java/io/cdap/plugin/db/SchemaReader.java

```diff
@@ -49,4 +48,22 @@
    * @throws SQLException
    */
   Schema getSchema(ResultSetMetaData metadata, int index) throws SQLException;
+
+  /**
+   * Given a sql metadata indicates if sql column must be ignored and not included in the output schema
+   * @param metadata resultSet metadata
+   * @param index sql column index
+   * @return 'true' if sql column must not included in the output schema, 'false' otherwise
+   * @throws SQLException
+   */
+  boolean ignoreColumn(ResultSetMetaData metadata, int index) throws SQLException;
```

Comment:
Allows to not include a column in the inferred schema. Thus we can support different data types for Sink and Source plugins.

For example, in MS SQL `TIMESTAMP` is the synonym for the `ROWVERSION` data type, values of which are automatically generated and can not be inserted or updated. Therefore `TIMESTAMP` can not be supported by Sink plugin.

`TIMESTAMP` reported as non-nullable column by JDBC connector, so inferred schema will be also non-nullable for this field, requiring to set a value. By ignoring this column we will avoid schema validation failure.

Reply:
Added to the javadoc.